### PR TITLE
RSE-722 Fix: Expand Timeout Variable on Scheduled Job

### PR DESCRIPTION
**Pull request by DarwisNarvaezDev**
# RSE-722 Fix: Expand Timeout Variable on Scheduled Job
The 'timeout' variable wont get expanded when the job is triggered by a schedule.

## The Problem
The timeout was taken literally from the DB

## The Solution
Expanding the variable programmatically as the 'user ran' process workflow.

# To QA :heavy_check_mark: 
1. Create a job with an option named "timeout" and default value of 5 and
2. A step which lasts 15 seconds then
3. Go to "other" in job definition and input '${option.timeout}' in 'Timeout' field also
4. Schedule it to run each minute and save the job

:checkered_flag: Expected behavior: the job fails in 5 seconds after the schedule triggers it.

### CODE REVIEW

File: rundeckapp/grails-app/jobs/rundeck/quartzjobs/ExecutionJob.groovy [hunk 1]
```diff
@@ -338,13 +336,23 @@ class ExecutionJob implements InterruptableJob {
...
         }
         if (!initMap.authContext) {
             throw new RuntimeException("authContext could not be determined")
         }
         return initMap
     }
 
+    @CompileStatic
+    private long expandTimeout(RunContext initMap){
+        def jobArguments=FrameworkService.parseOptsFromString(initMap.execution?.argString)
```
Comment (alexander-variacode) [inline rundeckapp/grails-app/jobs/rundeck/quartzjobs/ExecutionJob.groovy:351]: This is a property comes from framework.properties?

**Review by alexander-variacode — COMMENTED** [review]
Add more test options like -1, null, 0

**Review by alexander-variacode — APPROVED** [review]
LGTM
